### PR TITLE
Code and related files for the dependency injection demos.

### PR DESCRIPTION
**Pull request by reymalahay**
@eugenp i recently made more changes to the code, as per editorial discussion with @atheedom

### CODE REVIEW

File: dependency-injection/build.gradle [hunk 1]
```diff
@@ -0,0 +1,43 @@
+apply plugin: 'java'
```
Comment (pivovarit) [inline dependency-injection/build.gradle]: @eugenp is this ok to mix Gradle with Maven?

Comment (eugenp) [inline dependency-injection/build.gradle]: Not generally, unless we have a good reason to do so, I'd stick with Maven.

Comment (reymalahay) [inline dependency-injection/build.gradle]: @eugenp, should i delete the build.gradle file?

Comment (eugenp) [inline dependency-injection/build.gradle]: I haven't looked at the code in detail, but yes - basically, most projects here use Maven, so let's make sure this does the same.

Comment (reymalahay) [inline dependency-injection/build.gradle]: @eugenp, i'll do that now. closing this pull request.

Comment (eugenp) [inline dependency-injection/build.gradle]: You can update a PR, you don't always need to close it and do a new one.

Comment (reymalahay) [inline dependency-injection/build.gradle]: @eugenp, thank you for the advice.

**Pull request closed without merging**